src/redux/books/books.js

```diff
@@ -0,0 +1,28 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REM_BOOK = 'REM_BOOK';
+
+let nextId = 0;
+
+export function addBookObject(title, author) {
+  nextId += 1;
+  return {
+    type: ADD_BOOK, id: nextId, title, author,
+  };
+}
+
+export function removeBookObject(id) {
+  return { type: REM_BOOK, id };
+}
+
+const bookReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, { id: action.id, title: action.title, author: action.author }];
```

Comment:
As a result here you can simply pass `action.payload` since payload is already an object that contains all the information pertaining to a book. The action parameter expects to receive a single object or array of objects, typically a response JSON from an API.
```
return [...state, action.payload];
```

Reply:
Implemented 👍